[BUG] test_hash_reduction_decimal_overflow_sum[30] failed OOM in integration tests

**Describe the bug**
related to #4272 

FAILED ../../src/main/python/hash_aggregate_test.py::test_hash_reduction_decimal_overflow_sum[30]

```
java.lang.OutOfMemoryError: Could not allocate native memory: std::bad_alloc: out_of_memory: RMM failure at:/home/jenkins/agent/workspace/jenkins-cudf_nightly-dev-github-556-cuda11/cpp/build/_deps/rmm-src/include/rmm/mr/device/arena_memory_resource.hpp:157: Maximum pool size exceeded
[2021-12-07T06:04:57.784Z] 	at ai.rapids.cudf.ColumnView.binaryOpVS(Native Method)
[2021-12-07T06:04:57.784Z] 	at ai.rapids.cudf.ColumnView.binaryOp(ColumnView.java:1205)
[2021-12-07T06:04:57.784Z] 	at ai.rapids.cudf.ColumnView.binaryOp(ColumnView.java:1195)
[2021-12-07T06:04:57.784Z] 	at ai.rapids.cudf.BinaryOperable.div(BinaryOperable.java:171)
[2021-12-07T06:04:57.784Z] 	at org.apache.spark.sql.rapids.GpuDecimalSumHighDigits.$anonfun$columnarEval$10(AggregateFunctions.scala:720)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
[2021-12-07T06:04:57.784Z] 	at org.apache.spark.sql.rapids.GpuDecimalSumHighDigits.withResource(AggregateFunctions.scala:690)
[2021-12-07T06:04:57.784Z] 	at org.apache.spark.sql.rapids.GpuDecimalSumHighDigits.$anonfun$columnarEval$9(AggregateFunctions.scala:719)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
[2021-12-07T06:04:57.784Z] 	at org.apache.spark.sql.rapids.GpuDecimalSumHighDigits.withResource(AggregateFunctions.scala:690)
[2021-12-07T06:04:57.784Z] 	at org.apache.spark.sql.rapids.GpuDecimalSumHighDigits.$anonfun$columnarEval$8(AggregateFunctions.scala:718)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
[2021-12-07T06:04:57.784Z] 	at org.apache.spark.sql.rapids.GpuDecimalSumHighDigits.withResource(AggregateFunctions.scala:690)
[2021-12-07T06:04:57.784Z] 	at org.apache.spark.sql.rapids.GpuDecimalSumHighDigits.columnarEval(AggregateFunctions.scala:713)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:34)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.GpuExpressionsUtils$.columnarEvalToColumn(GpuExpressions.scala:94)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.GpuProjectExec$.projectSingle(basicPhysicalOperators.scala:102)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.GpuProjectExec$.$anonfun$project$1(basicPhysicalOperators.scala:109)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.$anonfun$safeMap$1(implicits.scala:162)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.$anonfun$safeMap$1$adapted(implicits.scala:159)
[2021-12-07T06:04:57.784Z] 	at scala.collection.immutable.List.foreach(List.scala:392)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.safeMap(implicits.scala:159)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.RapidsPluginImplicits$AutoCloseableProducingSeq.safeMap(implicits.scala:194)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.GpuProjectExec$.project(basicPhysicalOperators.scala:109)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator$AggHelper.preProcess(aggregate.scala:698)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator.$anonfun$computeAggregate$1(aggregate.scala:776)
[2021-12-07T06:04:57.784Z] 	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator.withResource(aggregate.scala:180)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator.com$nvidia$spark$rapids$GpuHashAggregateIterator$$computeAggregate(aggregate.scala:773)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator.$anonfun$aggregateInputBatches$1(aggregate.scala:284)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator.$anonfun$aggregateInputBatches$1$adapted(aggregate.scala:282)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator.withResource(aggregate.scala:180)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator.aggregateInputBatches(aggregate.scala:282)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator.$anonfun$next$2(aggregate.scala:237)
[2021-12-07T06:04:57.785Z] 	at scala.Option.getOrElse(Option.scala:189)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:234)
[2021-12-07T06:04:57.785Z] 	at com.nvidia.spark.rapids.GpuHashAggregateIterator.next(aggregate.scala:180)
[2021-12-07T06:04:57.785Z] 	at org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase$$anon$1.partNextBatch(GpuShuffleExchangeExecBase.scala:288)
[2021-12-07T06:04:57.785Z] 	at org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExecBase$$anon$1.hasNext(GpuShuffleExchangeExecBase.scala:304)
[2021-12-07T06:04:57.785Z] 	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:132)
[2021-12-07T06:04:57.785Z] 	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
[2021-12-07T06:04:57.785Z] 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
[2021-12-07T06:04:57.785Z] 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
[2021-12-07T06:04:57.785Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:127)
[2021-12-07T06:04:57.785Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:446)
[2021-12-07T06:04:57.785Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
[2021-12-07T06:04:57.785Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:449)
[2021-12-07T06:04:57.785Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2021-12-07T06:04:57.785Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2021-12-07T06:04:57.785Z] 	at java.lang.Thread.run(Thread.java:748)
```

--- a/jenkins/spark-tests.sh
+++ b/jenkins/spark-tests.sh
@@ -221,7 +221,7 @@ if [[ $TEST_MODE == "ALL" || $TEST_MODE == "IT_ONLY" ]]; then
     # --halt "now,fail=1": exit when the first job fail, and kill running jobs.
     #                      we can set it to "never" and print failed ones after finish running all tests if needed
     # --group: print stderr after test finished for better readability
-    parallel --group --halt "now,fail=1" -j"${PARALLELISM}" run_test ::: $tests
+    parallel --group --halt "now,fail=1" -j"${PARALLELISM}" run_test ::: hash_aggregate_test.py
   else
     run_test all
   fi

--- a/jenkins/spark-tests.sh
+++ b/jenkins/spark-tests.sh
@@ -221,7 +221,7 @@ if [[ $TEST_MODE == "ALL" || $TEST_MODE == "IT_ONLY" ]]; then
     # --halt "now,fail=1": exit when the first job fail, and kill running jobs.
     #                      we can set it to "never" and print failed ones after finish running all tests if needed
     # --group: print stderr after test finished for better readability
-    parallel --group --halt "now,fail=1" -j"${PARALLELISM}" run_test ::: hash_aggregate_test.py
+    parallel -u --halt "now,fail=1" -j"${PARALLELISM}" run_test ::: hash_aggregate_test.py test_hash_reduction_decimal_overflow_sum
   else
     run_test all
   fi

--- a/jenkins/spark-tests.sh
+++ b/jenkins/spark-tests.sh
@@ -215,7 +215,7 @@ if [[ $TEST_MODE == "ALL" || $TEST_MODE == "IT_ONLY" ]]; then
   # integration tests
   if [[ $PARALLEL_TEST == "true" ]] && [ -x "$(command -v parallel)" ]; then
     # put most time-consuming tests at the head of queue
-    time_consuming_tests="join_test.py generate_expr_test.py parquet_write_test.py"
+    time_consuming_tests="join_test.py hash_aggregate_test.py parquet_write_test.py"
     tests_list=$(find "$SCRIPT_PATH"/src/main/python/ -name "*_test.py" -printf "%f ")
     tests=$(echo "$time_consuming_tests $tests_list" | tr ' ' '\n' | awk '!x[$0]++' | xargs)
     # --halt "now,fail=1": exit when the first job fail, and kill running jobs.

--- a/jenkins/spark-tests.sh
+++ b/jenkins/spark-tests.sh
@@ -139,14 +139,16 @@ export SEQ_CONF="--executor-memory 16G \
 
 # currently we hardcode the parallelism and configs based on our CI node's hardware specs,
 # we can make it dynamically generated if this script is going to be used in other scenarios in the future
+PARALLELISM=${PARALLELISM:-'4'}
+MEMORY_FRACTION=$(python -c "print(1/($PARALLELISM + 1))")
 export PARALLEL_CONF="--executor-memory 4G \
 --total-executor-cores 2 \
 --conf spark.executor.cores=2 \
 --conf spark.task.cpus=1 \
 --conf spark.rapids.sql.concurrentGpuTasks=2 \
---conf spark.rapids.memory.gpu.allocFraction=0.15 \
 --conf spark.rapids.memory.gpu.minAllocFraction=0 \
---conf spark.rapids.memory.gpu.maxAllocFraction=0.15"
+--conf spark.rapids.memory.gpu.allocFraction=${MEMORY_FRACTION} \
+--conf spark.rapids.memory.gpu.maxAllocFraction=${MEMORY_FRACTION}"
 
 export CUDF_UDF_TEST_ARGS="--conf spark.rapids.memory.gpu.allocFraction=0.1 \
 --conf spark.rapids.memory.gpu.minAllocFraction=0 \
@@ -219,7 +221,7 @@ if [[ $TEST_MODE == "ALL" || $TEST_MODE == "IT_ONLY" ]]; then
     # --halt "now,fail=1": exit when the first job fail, and kill running jobs.
     #                      we can set it to "never" and print failed ones after finish running all tests if needed
     # --group: print stderr after test finished for better readability
-    parallel --group --halt "now,fail=1" -j5 run_test ::: $tests
+    parallel --group --halt "now,fail=1" -j"${PARALLELISM}" run_test ::: $tests
   else
     run_test all
   fi

--- a/jenkins/spark-tests.sh
+++ b/jenkins/spark-tests.sh
@@ -140,7 +140,7 @@ export SEQ_CONF="--executor-memory 16G \
 # currently we hardcode the parallelism and configs based on our CI node's hardware specs,
 # we can make it dynamically generated if this script is going to be used in other scenarios in the future
 PARALLELISM=${PARALLELISM:-'4'}
-MEMORY_FRACTION=$(python -c "print(1/($PARALLELISM + 1))")
+MEMORY_FRACTION=$(python -c "print(1/($PARALLELISM + 0.2))")
 export PARALLEL_CONF="--executor-memory 4G \
 --total-executor-cores 2 \
 --conf spark.executor.cores=2 \
@@ -221,7 +221,7 @@ if [[ $TEST_MODE == "ALL" || $TEST_MODE == "IT_ONLY" ]]; then
     # --halt "now,fail=1": exit when the first job fail, and kill running jobs.
     #                      we can set it to "never" and print failed ones after finish running all tests if needed
     # --group: print stderr after test finished for better readability
-    parallel -u --halt "now,fail=1" -j"${PARALLELISM}" run_test ::: hash_aggregate_test.py test_hash_reduction_decimal_overflow_sum
+    parallel --group --halt "now,fail=1" -j"${PARALLELISM}" run_test ::: $tests
   else
     run_test all
   fi

